maps/src/main/java/org/odk/collect/maps/layers/DirectoryReferenceLayerRepository.kt

```diff
@@ -27,7 +28,15 @@ class DirectoryReferenceLayerRepository(
         }
     }
 
-    private fun getAllFilesWithDirectory() = directoryPaths.flatMap { dir ->
+    override fun getSharedLayersDirPath(): String {
```

Comment:
I think the actual directories can stay encapsulated within the implementation here (and not be exposed from the `ReferenceLayerRepository` interface). Instead of grabbing the paths and then passing to `importNewLayers`, we could add a method to `ReferenceLayerRepository`:

```kotlin
fun addLayer(file: File, shared: Boolean)
```

Then, `importNewLayers` just needs passed `shared` rather the UI needing to know about the file paths.

Reply:
Done.